CMakeLists.txt

```diff
@@ -234,6 +234,16 @@ if(ENABLE_SERVICES)
   endif()
 endif()
 
+if (ENABLE_CLANG_TIDY)
+  find_program(CLANG_TIDY_COMMAND NAMES clang-tidy)
+  if(NOT CLANG_TIDY_COMMAND)
+    message(FATAL_ERROR "ENABLE_CLANG_TIDY is ON but clang-tidy is not found!")
+  else()
+    message(STATUS "Found clang-tidy at ${CLANG_TIDY_COMMAND}")
+    set(CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_COMMAND};--warnings-as-errors=*")
+  endif()
+endif()
```

Comment:
Running clang-tidy on full codebase on every PR is horribly slow. I would keep approach via `scripts/clang-tidy-only-diff.sh`. Still, it is a good idea to run it on all codebase every year or every release...